pkg/adapter/nimbus-k8tls/builder/builder.go

```diff
@@ -122,6 +151,35 @@ func cronJobForEnsureTls(schedule string, externalAddresses ...string) (*batchv1
 		},
 	}
 
+	// Fetch the elasticsearch password secret. If the secret is present, set TTLSecondsAfterFinished and reference the secret in the cronjob templateZ
+	var elasticsearchPasswordSecret corev1.Secret
+	err := ctx.Value(common.K8sClientKey).(client.Client).Get(ctx, client.ObjectKey{Namespace: ctx.Value(common.NamespaceNameKey).(string), Name: "elasticsearch-password"}, &elasticsearchPasswordSecret)
```

Comment:
I think we should simplify this call as follows.
```suggestion
	err := k8sClient.Get(ctx, client.ObjectKey{Namespace: NamespaceName, Name: "elasticsearch-password"}, &elasticsearchPasswordSecret)
```